frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDetailsVersionContext.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import usePipelineVersionById from '~/concepts/pipelines/apiHooks/usePipelineVersionById';
+import { getPipelineVersionResourceRef } from '~/concepts/pipelines/content/tables/utils';
+import {
+  PipelineRunJobKF,
+  PipelineRunKF,
+  PipelineVersionKF,
+  ResourceReferenceKF,
+} from '~/concepts/pipelines/kfTypes';
+
+type PipelineRunDetailsVersionContextState = {
+  version: PipelineVersionKF | null;
+  loaded: boolean;
+  error?: Error;
+  versionRef?: ResourceReferenceKF;
+};
+
+export const PipelineRunDetailsVersionContext =
+  React.createContext<PipelineRunDetailsVersionContextState>({
+    version: null,
+    loaded: false,
+    error: undefined,
+    versionRef: undefined,
+  });
+
+type PipelineRunDetailsVersionContextProps = {
+  children: React.ReactNode;
+  resource?: PipelineRunJobKF | PipelineRunKF | null;
+};
+
+const PipelineRunDetailsVersionContextProvider: React.FC<PipelineRunDetailsVersionContextProps> = ({
+  children,
+  resource,
+}) => {
+  const versionRef = getPipelineVersionResourceRef(resource);
```

Comment:
It looks like this context is only accessed in 1 place, where this version data is needed, I'm wondering if it makes more sense to simply store the "run" object for this context and instead call it `PipelineRunDetailsContext`. Since I think keeping track of a run from a couple different locations is its main purpose.